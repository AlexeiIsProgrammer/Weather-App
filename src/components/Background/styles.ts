import styled from 'styled-components';

export const BackgroundContainer = styled.main<{
  $backgroundImage: string;
  $isLoaded: boolean;
}>`
  background-image: ${(props) => props.$backgroundImage};
  height: 100%;
  width: 100%;
  z-index: -1;
  position: fixed;
  left: 0px;
  top: 0px;
  opacity: ${(props) => (props.$isLoaded ? 1 : 0)};
  transition: 1s ease;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
`;
