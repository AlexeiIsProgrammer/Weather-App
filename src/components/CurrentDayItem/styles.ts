import { styled } from 'styled-components';

import {
  DayDate, DayImage, DayTemp, DayWrapper,
} from '../DayItem/styles';

export const CurrentDayWrapper = styled(DayWrapper)`
  cursor: default;
  display: flex;
  flex-direction: row-reverse;
  justify-content: space-between;
  padding: ${(props) => props.theme.padding.medium};
  gap: ${(props) => props.theme.gap.small};
  margin-bottom: ${(props) => props.theme.margin.medium};
`;

export const CurrentDayDate = styled(DayDate)`
  font-size: ${(props) => props.theme.fontSizes.em.large};

  @media screen and (max-width: 450px) {
    font-size: ${(props) => props.theme.fontSizes.em.medium};
  }
`;
export const CurrentDayTemp = styled(DayTemp)`
  font-weight: ${(props) => props.theme.fontWeights.bold};
  font-size: ${(props) => props.theme.fontSizes.em.large};

  @media screen and (max-width: 450px) {
    font-size: ${(props) => props.theme.fontSizes.em.medium};
  }
`;
export const CurrentDayImage = styled(DayImage)`
  width: ${(props) => props.theme.iconSize.large};
  height: ${(props) => props.theme.iconSize.large};

  @media screen and (max-width: 450px) {
    width: ${(props) => props.theme.iconSize.small};
    height: ${(props) => props.theme.iconSize.small};
  }
`;

export const CurrentDayImageImg = styled.img`
  width: ${(props) => props.theme.iconSize.large};
  height: ${(props) => props.theme.iconSize.large};

  @media screen and (max-width: 450px) {
    width: ${(props) => props.theme.iconSize.small};
    height: ${(props) => props.theme.iconSize.small};
  }
`;
