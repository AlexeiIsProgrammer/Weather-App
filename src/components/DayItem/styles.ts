import { styled } from 'styled-components';

export const DayWrapper = styled.div<{ $active?: boolean }>`
  color: ${(props) =>
    props.$active ? props.theme.colors.white : props.theme.colors.black};
  background-color: ${(props) =>
    props.$active ? props.theme.colors.blue : props.theme.colors.white};

  padding: ${(props) => props.theme.padding.small};
  border: 1px solid ${(props) => props.theme.colors.black};
  border-radius: ${(props) => props.theme.borderRadius.medium};
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  flex: 1;
  justify-content: center;
  gap: ${(props) => props.theme.gap.small};
  transform: translateY(0px);
  transition: ${(props) => props.theme.transition.slow};
  cursor: pointer;

  @media (hover: hover) {
    &:not(&__current):hover {
      transform: translateY(-10px);
      box-shadow: 0px 0px 5px ${(props) => props.theme.colors.black};
    }
  }

  &:active {
    transform: translateY(-10px);
    box-shadow: 0px 0px 5px ${(props) => props.theme.colors.black};
  }
`;

export const DayDate = styled.p``;
export const DayTemp = styled.span`
  font-weight: ${(props) => props.theme.fontWeights.bold};
`;
export const DayImage = styled.div`
  width: ${(props) => props.theme.iconSize.small};
  height: ${(props) => props.theme.iconSize.small};
`;
