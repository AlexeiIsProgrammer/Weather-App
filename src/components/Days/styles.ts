import { styled } from 'styled-components';

export const DaysList = styled.ul`
  display: flex;
  flex-direction: row;
  gap: 20px;
  justify-content: space-between;

  @media screen and (max-width: 991px) {
    overflow-x: auto;
  }
`;

export const ErrorMessage = styled.h1`
  text-align: center;
`;
