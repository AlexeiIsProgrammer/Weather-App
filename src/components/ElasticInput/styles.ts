import { styled } from 'styled-components';

export const ElasticInputContainer = styled.div`
  position: relative;
  flex: 1;
`;
