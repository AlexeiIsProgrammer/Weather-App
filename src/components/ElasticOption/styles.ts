import styled from 'styled-components';

export const ElasticOptionContainer = styled.button`
  cursor: pointer;

  padding: ${(props) => props.theme.padding.small};
  background-color: ${(props) => props.theme.colors.white};
  border-radius: ${(props) => props.theme.borderRadius.medium};
  border: 1px solid ${(props) => props.theme.colors.black};
  color: ${(props) => props.theme.colors.black};
  transition: ${(props) => props.theme.transition.fast};

  &:hover {
    opacity: 0.5;
  }
`;
