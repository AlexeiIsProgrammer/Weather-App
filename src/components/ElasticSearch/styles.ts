import { styled } from 'styled-components';

export const ElasticSearchContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 10px;
  width: 100%;

  @media (max-width: 440px) {
    flex-direction: column;
  }
`;
