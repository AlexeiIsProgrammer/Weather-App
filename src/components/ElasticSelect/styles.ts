import styled from 'styled-components';

export const SelectResultState = styled.p`
  text-align: ${(props) => props.theme.textAlign.center};
  width: ${(props) => props.theme.width.full};
`;

export const ElasticSelectContainer = styled.div`
  position: absolute;
  box-sizing: border-box;

  display: flex;
  gap: 20px;
  flex-direction: row;
  flex-wrap: wrap;

  z-index: 1;
  top: calc(100% + 10px);
  left: 0;
  overflow-y: ${(props) => props.theme.overflow.auto};

  max-height: ${(props) => props.theme.maxHeight.select};
  width: ${(props) => props.theme.width.full};
  padding: ${(props) => props.theme.padding.medium};
  padding-right: ${(props) => props.theme.padding.large};
  background-color: ${(props) => props.theme.colors.white};
  border-radius: ${(props) => props.theme.borderRadius.medium};
  border: 1px solid ${(props) => props.theme.colors.black};
  color: ${(props) => props.theme.colors.black};
  font-size: ${(props) => props.theme.fontSizes.em.small};
`;

export const CloseSelectButton = styled.button`
  position: absolute;
  cursor: pointer;
  width: 20px;
  height: 20px;

  right: 10px;
  top: 10px;

  &::after,
  &::before {
    content: '';
    position: absolute;

    width: 20px;
    height: 2px;

    background-color: black;
  }

  &::after {
    transform: translateY(250%) rotate(-45deg);
  }

  &::before {
    transform: translateY(250%) rotate(45deg);
  }
`;
