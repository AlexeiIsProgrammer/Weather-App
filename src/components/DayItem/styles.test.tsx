import React from 'react';
import renderer from 'react-test-renderer';

import 'jest-styled-components';
import { DayWrapper } from './styles';

test('should change color of inactive element', () => {
  const wrapper = renderer.create(<DayWrapper $active={false} />).toJSON();

  expect(wrapper).toHaveStyleRule('color', 'black');
  expect(wrapper).toHaveStyleRule('background-color', 'white');
});

test('should change color of active element', () => {
  const wrapper = renderer.create(<DayWrapper $active />).toJSON();

  expect(wrapper).toHaveStyleRule('color', 'white');
  expect(wrapper).toHaveStyleRule('background-color', 'blue');
});
