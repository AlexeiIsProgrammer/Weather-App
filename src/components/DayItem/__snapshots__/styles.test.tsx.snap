// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change color of active element 1`] = `
.c0 {
  color: white;
  background-color: blue;
  padding: 5px;
  border: 1px solid black;
  border-radius: 10px;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  flex: 1;
  justify-content: center;
  gap: 5px;
  transform: translateY(0px);
  transition: 0.5s ease;
  cursor: pointer;
}

.c0:active {
  transform: translateY(-10px);
  box-shadow: 0px 0px 5px #000;
}

@media (hover: hover) {
  .c0:not(.c0__current):hover {
    transform: translateY(-10px);
    box-shadow: 0px 0px 5px #000;
  }
}

<div
  className="c0"
/>
`;

exports[`should change color of inactive element 1`] = `
.c0 {
  color: black;
  background-color: white;
  padding: 5px;
  border: 1px solid black;
  border-radius: 10px;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  flex: 1;
  justify-content: center;
  gap: 5px;
  transform: translateY(0px);
  transition: 0.5s ease;
  cursor: pointer;
}

.c0:active {
  transform: translateY(-10px);
  box-shadow: 0px 0px 5px #000;
}

@media (hover: hover) {
  .c0:not(.c0__current):hover {
    transform: translateY(-10px);
    box-shadow: 0px 0px 5px #000;
  }
}

<div
  className="c0"
/>
`;
